### DISCUSSION TITLE
How do I configure buggregator for a vanilla PHP app?

### DISCUSSION THREAD

**Discussion by nsrosenqvist** [buggregator/server#206] [Q&A] [answered]
The documentation is quite sparse on information on how one would go about configuring xhprof and vardumper if you're just running a vanilla PHP application. The "configuration" sections probably only apply to the packages for Laravel and Spiral, unless I'm mistaken. A what levels of the stack would I need to integrate these systems and how would one go about it? I can't find much good documentation on xhprof

**Comment by butschster** [comment, marked as answer]
Hey! To configure var dumper for a vanilla PHP application, you just need to use `$_SERVER` variables. Check this out: [Var Dumper Configuration](https://docs.buggregator.dev/config/var-dumper.html#configuration).

Here's a quick example:

```php
// Plain PHP
$_SERVER['VAR_DUMPER_FORMAT'] = 'server';
$_SERVER['VAR_DUMPER_SERVER'] = '127.0.0.1:9912';
```

You can also set these variables using a bash script.

```bash
#!/bin/bash

# Set environment variables for var dumper
export VAR_DUMPER_FORMAT="server"
export VAR_DUMPER_SERVER="127.0.0.1:9912"
```


For xhprof, it's a good idea to add a setup section for vanilla PHP applications. Thanks for the suggestion! It's not too hard. Just use the `spiral-packages/profiler` composer package.

Here's an example:

```php
use SpiralPackages\Profiler\Profiler;
use SpiralPackages\Profiler\Driver\DriverInterface;
use SpiralPackages\Profiler\DriverFactory;
use SpiralPackages\Profiler\Storage\StorageInterface;
use SpiralPackages\Profiler\Storage\WebStorage;
use Symfony\Component\HttpClient\NativeHttpClient;

$storage = new WebStorage(
    new NativeHttpClient(),
    'http://127.0.0.1/api/profiler/store',
);

$driver = DriverFactory::detect();

$profiler = new Profiler(
    storage: $storage, 
    driver: $driver, 
    appName: 'My super app', 
    tags: [
        // global tags
        'env' => 'local'
    ]
);

$profiler->start(ignoredFunctions: []);

// Here is your code you want to profile

$profiler->end(tags: [
    // Tags for specific requests
]);
```

Hope this helps! If you need more details, let me know!

---

It would be awesome if you could create a [PR to the docs repository](https://github.com/buggregator/docs) to improve the documentation for vanilla PHP users. Thanks a lot!

**Reply by nsrosenqvist** [reply]
Thanks! I'm giving it a try and will see if I get it up and running :)

**Reply by nsrosenqvist** [reply]
@butschster Any idea why every property included under "extra" would be formatted as its own object (redacted two lines and cut the list short)?
<img width="793" alt="Screenshot 2024-06-24 at 13 46 39" src="https://github.com/buggregator/server/assets/1303475/cb062965-72af-4a73-a8ca-60bbc42694c7">

**Reply by nsrosenqvist** [reply]
@butschster is this a bug or a misconfiguration on my end?

**Reply by butschster** [reply]
Hey! Could you send me JSON payload for your monolog log? 
![image](https://github.com/buggregator/server/assets/773481/4551b95c-f96e-4f7c-835e-717b9ead0c52)

I need to check your log contenxt.

**Reply by nsrosenqvist** [reply]
@butschster Thanks for your help in debugging the issue, here is the payload (with redactions):
```json
{
    "uuid": "019059fe-9284-71fe-9e47-a392664a8c87",
    "project": null,
    "type": "monolog",
    "payload": {
        "channel": "application",
        "context": {
            "cool": "heck yeah",
            "here": "where?"
        },
        "datetime": "2024-06-27T15:59:34.201624+02:00",
        "extra": {
            "callType": null,
            "class": null,
            "file": "[REDACTED]",
            "function": "[REDACTED]",
            "http_method": "GET",
            "instance": "[REDACTED]",
            "ip": "[REDACTED]",
            "line": 514,
            "referrer": null,
            "server": "[REDACTED]",
            "url": "[REDACTED]"
        },
        "level": 200,
        "level_name": "INFO",
        "message": "Hello world"
    },
    "timestamp": 1719496774.2804
}
```

**Reply by nsrosenqvist** [reply]
@butschster I'll go ahead and file an issue unless you have something else to add regarding this?

**Reply by nsrosenqvist** [reply]
@butschster https://github.com/buggregator/frontend/issues/222

Let me know if I submitted it against the wrong repo!
